app/importers/file_importers/streaming_csv_transformer.rb

```diff
@@ -0,0 +1,73 @@
+require 'csv'
+
+# This implements:
+#   the Transformer interface {transform, pre_cleanup_csv_size}
+#   our ad-hoc CSV data interface {each_with_index, size}
+class StreamingCsvTransformer
+  def initialize(options = {})
+    @headers = options.key?(:headers) ? options[:headers] : true
+    @total_rows_count = nil
+    @processed_rows_count = nil
+  end
+
+  # Performs whole-file transformations first
+  def transform(csv_string)
+    @csv_string = create_sanitized_csv_string(csv_string)
+    self
+  end
+
+  # This is only valid after data is read with #each_with_index
+  def pre_cleanup_csv_size
+    @total_rows_count
+  end
+
+  # Stream each line of the CSV and parse it.
+  # This includes three different types of transformations to replicate
+  # the behavior of CsvTransformer for now.  These could be pushed out to
+  # callers in the future.  These are: nil_converter for each field, discarding rows with
+  # dirty_data? and parsing particular field names into Ruby dates.
+  def each_with_index(&block)
+    @total_rows_count = 0
+    @processed_rows_count = 0
+
+    CSV.parse(@csv_string, {
+      headers: @headers,
+      header_converters: :symbol,
+      encoding: 'binary:UTF-8',
+      converters: lambda { |h| nil_converter(h) }
+    }) do |row, index|
+      @total_rows_count = @total_rows_count + 1
+      cleaner = CsvRowCleaner.new(row)
+      next if cleaner.dirty_data?
```

Comment:
This is a lot of processing, that from the perspective of the importers is hidden inside the call to `CsvDownload` which sounds like it'd be a pretty simple operation

 I think ideally we'd push this out to importer classes; looking at production logs for "rows of data in" to see how much this is used it only impacts attendance import.  If this PR lands, we could factor this out there, and then it seems there's no need to be running these checks for other importers.  Nothing to do now, just noting it since I didn't remember about this.